src/hooks/useWalletBookmarkedProjects.ts

```diff
@@ -0,0 +1,53 @@
+import { useSupabaseClient } from '@supabase/auth-helpers-react'
+import { useEffect, useState } from 'react'
+import { Database } from 'types/database.types'
+
+import { useDBProjectsQuery } from './useProjects'
+
+export function useWalletBookmarkedProjects({
+  wallet,
+}: {
+  wallet: string | undefined
+}) {
+  const [createdAtIds, setCreatedAtIds] = useState<{ [id: string]: number }>({})
+
+  const supabase = useSupabaseClient<Database>()
+
+  useEffect(() => {
+    supabase
+      .from('user_bookmarks')
+      .select('*')
+      .eq('wallet', wallet?.toLowerCase())
+      .order('created_at', { ascending: false })
+      .then(data =>
+        setCreatedAtIds(
+          data.data?.reduce(
+            (acc, curr) => ({
+              ...acc,
+              [curr.project]: new Date(curr.created_at).valueOf(),
+            }),
+            {},
+          ) ?? {},
+        ),
+      )
+  }, [supabase, wallet])
+
+  const ids = Object.keys(createdAtIds ?? {})
+
+  const { data: projects, isLoading } = useDBProjectsQuery(
+    ids.length
+      ? {
+          ids,
+          pageSize: 1000,
+        }
+      : null,
+  )
```

Comment:
**suggestion** I think we can get all of this happening in the one query with a `join`. Not tested but something like this:

```
SELECT * FROM projects
WHERE user_bookmarks.wallet = {wallet_address}
INNER JOIN user_bookmarks ON user_bookmarks.project_id = projects.id
ORDER BY user_bookmarks.created_at
```

Reply:
ohhh thank you, this is what i needed. couldn't remember how to do this earlier

Reply:
@tomquirk shiiiiit just figured out some neat stuff 
<img width="565" alt="image" src="https://github.com/jbx-protocol/juice-interface/assets/79433522/e89e7fd5-5547-4578-86ca-0758ebaadef8">

Reply:
yooo. clean af, look at thattt